code/game/atoms.dm

```diff
@@ -84,6 +84,9 @@
 		stack_trace("Warning: [src]([type]) initialized multiple times!")
 	initialized = TRUE
 
+	if(loc)
+		SEND_SIGNAL(loc, COMSIG_ATOM_CREATED, src) // Sends a signal that the new atom `src`, has been created at `loc`.
```

Comment:
The atom will most likely not be fully initialised here. Usually Initialize is overriden and the parent is called prior to the other code. Best would be to send this signal after Initialize is fully done and thus in the proc above calling this one.

Reply:
As far as I can tell the atom is spawned in the game world at this point, and my tests worked with having it here, but, I'm not sure if it will cause weird behavior. Guess I should put it at the end of the init proc, yeah.

Reply:
An atom can override it like this:
![image](https://user-images.githubusercontent.com/15887760/78139593-0e436380-7429-11ea-9118-fd9e8b8a5eca.png)

Currently the signal will go through before Link_Abduction_Equipment is called. So the atom is not yet fully initialized.

On line 87 of subsystem/atoms.dm in `InitAtom` the `Initialize` is called. If you send the signal after that call then you'll be sure it's Initialized. Can even use some logic there (it has some checks to see if the init actually worked)

Reply:
Yeah that's a good idea, I'm thinking this is a good spot:

![image](https://user-images.githubusercontent.com/42044220/78170756-0b407700-7419-11ea-9aef-e1b9edbb5867.png)

I believe if the code reaches this point, we know for sure the atom was initialized.

Reply:
Won't this code get called twice now?

Reply:
Yeah that was me being dumb and forgetting to remove the previous one.